modules/candidate_profile/php/module.class.inc

```diff
@@ -0,0 +1,88 @@
+<?php
+namespace LORIS\candidate_profile;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\StudyEntities\Candidate\CandID;
+
+/**
+ * Class module implements the basic LORIS module functionality
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class Module extends \Module
+{
+    /**
+     * The candidate_profile module overrides the default routing to ensure that
+     * /candidate_profile/123456 goes to the candidate_profile page for candidate
+     * 123456.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $candIDStr = ltrim($request->getURI()->getPath(), '/');
+        try {
+            $candID    = new CandID($candIDStr);
+            $candidate = \Candidate::singleton($candID);
+
+            $request = $request->withAttribute('candidate', $candidate);
+            $page    = $this->loadPage("candidate_profile");
+            return $page->process($request, $page);
+
+        } catch (\DomainException | \LORISException $e) {
+            // A LORISException means the \Candidate::singleton couldn't
+            // load the candidate (ie because the CandID doesn't exist
+            // in the database.) A \DomainException means the CandID format
+            // was invalid. In either case, let the parent handle the request
+            // and return a 404 if appropriate. (or it could be a js or css
+            // or other type of URL for the module..)
+            return parent::handle($request);
+        }
+    }
+
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return string
+     */
+    public function getLongName() : string
+    {
+        return "Candidate Profile";
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param string $type    The type of widgets to get
+     * @param \User  $user    The user widgets are being retrieved for.
+     * @param array  $options A type dependent list of options to provide
+     *                        to the widget.
+     *
+     * @return \LORIS\GUI\Widget[]
+     */
+    public function getWidgets(string $type, \User $user, array $options) : array
```

Comment:
I don't see that param `$options` being used in this PR. Can you explain why you added it? I'm not really able to work out what it's supposed to do based on the PHPDoc.

Reply:
It's passed at https://github.com/aces/Loris/pull/6092/files#diff-6dadba9ad9ecd0d6c9ebb445ce787335R83. This doesn't have any widgets, but other modules can use it to customize the widget in some way for the candidate (ie adding props or selecting data from the database.)